extensions/arc/src/ui/dashboards/controller/controllerDashboardOverviewPage.ts

```diff
@@ -127,11 +97,33 @@ export class ControllerDashboardOverviewPage extends DashboardPage {
 			ariaLabel: loc.arcResources
 		}).component();
 
-		const arcResourcesTableContainer = this.modelView.modelBuilder.divContainer()
-			.withItems([this._arcResourcesTable])
+		// Update loaded components with data
+		this.handleRegistrationsUpdated();
+		this.handleEndpointsUpdated();
+
+		// Assign the loading component after it has data
+		this._propertiesLoadingComponent.component = this._propertiesContainer;
```

Comment:
Why not just create the loading component here instead?

Reply:
Because the handle*Updated functions reference the loading components themselves.  They set .loading to true or false.  I wanted to re-use that logic in both the initial and refresh cases.

Reply:
And calling this.handleRegistrationsUpdated at the end instead doesn't work?

Reply:
That doesn't work.  I have another comment on this PR about this.   For reasons I don't understand, it matters that components have data before they're added to the loading component.  If they're updated after being added to the loading component, they don't show data.  The same thing occurred in the Postgres pages.  Easy to reproduce by switching the order.

Reply:
Interesting, don't remember running into that before. But I know that I've had plenty of issues with the loading component so not a huge surprise. Alright it's fine to leave it like this - possibly a slightly cleaner way to handle it but not enough of a concern to spend more time on.